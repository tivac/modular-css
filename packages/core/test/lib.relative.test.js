"use strict";

var path = require("path"),
    
    relative = require("../lib/relative.js");

describe("/lib", function() {
    describe("/relative.js", function() {
        it("should be a function", function() {
            expect(typeof relative).toBe("function");
        });

        it(`should export a "prefixed" property`, function() {
            expect(typeof relative.prefixed).toBe("function");
        });
        
        it("should resolve files against a specified cwd", function() {
            expect(
                relative(
                    process.cwd(),
                    require.resolve("./specimens/start.css")
                )
            )
            .toMatchSnapshot();
            
            expect(
                relative(
                    __dirname,
                    require.resolve("./specimens/start.css")
                )
            )
            .toMatchSnapshot();
        });

        it("should resolve files against a specified cwd w/ prefixes", function() {
            expect(
                relative.prefixed(
                    process.cwd(),
                    path.resolve("./packages/core/test/specimens/start.css")
                )
            )
            .toMatchSnapshot();
            
            expect(
                relative.prefixed(
                    path.resolve("./packages/core/test/specimens/"),
                    require.resolve("./lib.relative.test.js")
                )
            )
            .toMatchSnapshot();
        });
    });
});
