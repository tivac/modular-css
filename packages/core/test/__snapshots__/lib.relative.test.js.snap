// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/lib /relative.js should resolve files against a specified cwd 1`] = `"packages/core/test/specimens/start.css"`;

exports[`/lib /relative.js should resolve files against a specified cwd 2`] = `"specimens/start.css"`;

exports[`/lib /relative.js should resolve files against a specified cwd w/ prefixes 1`] = `"./packages/core/test/specimens/start.css"`;

exports[`/lib /relative.js should resolve files against a specified cwd w/ prefixes 2`] = `"../lib.relative.test.js"`;
